### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by hexonaut**
Improving the test coverage on the IRMs by comparing it to the default version. Ignoring stable rate borrow as we don't use it.

### CODE REVIEW

File: test/InterestRateStrategyBase.t.sol [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        // Disable stable borrow stuff
+        liquidityAvailable       = bound(liquidityAvailable, 0, MAX_TOKEN_AMOUNT);
+        params.unbacked          = bound(params.unbacked, 0, MAX_TOKEN_AMOUNT);
+        params.liquidityAdded    = bound(params.liquidityAdded, 0, MAX_TOKEN_AMOUNT);
+        params.liquidityTaken    = bound(params.liquidityTaken, 0, liquidityAvailable + params.liquidityAdded);
+        params.totalStableDebt   = 0;
+        params.totalVariableDebt = bound(params.totalVariableDebt, 0, MAX_TOKEN_AMOUNT);
+        params.reserveFactor     = bound(params.reserveFactor, 0, 100_00);
+        params.reserve           = address(asset);
+        params.aToken            = makeAddr("aToken");
```
Comment (lucas-manuel) [inline test/InterestRateStrategyBase.t.sol]:
```suggestion
        liquidityAvailable       = bound(liquidityAvailable,    0, MAX_TOKEN_AMOUNT);
        params.unbacked          = bound(params.unbacked,       0, MAX_TOKEN_AMOUNT);
        params.liquidityAdded    = bound(params.liquidityAdded, 0, MAX_TOKEN_AMOUNT);
        params.liquidityTaken    = bound(params.liquidityTaken, 0, liquidityAvailable + params.liquidityAdded);
        params.totalStableDebt   = 0;
        params.totalVariableDebt = bound(params.totalVariableDebt, 0, MAX_TOKEN_AMOUNT);
        params.reserveFactor     = bound(params.reserveFactor,     0, 100_00);
        params.reserve           = address(asset);
        params.aToken            = makeAddr("aToken");
```

File: test/InterestRateStrategyBase.t.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            uint256 expectedVariableBorrowRate
+        ) = defaultInterestStrategy.calculateInterestRates(params);
+        (
+            uint256 actualLiquidityRate,
+            ,
+            uint256 actualVariableBorrowRate
+        ) = interestStrategy.calculateInterestRates(params);
+
+        assertEq(expectedLiquidityRate, actualLiquidityRate, "liquidity rate mismatch");
+        assertEq(expectedVariableBorrowRate, actualVariableBorrowRate, "variable borrow rate mismatch");
```
Comment (lucas-manuel) [inline test/InterestRateStrategyBase.t.sol:70]:
```suggestion
        assertEq(expectedLiquidityRate,      actualLiquidityRate,      "liquidity rate mismatch");
        assertEq(expectedVariableBorrowRate, actualVariableBorrowRate, "variable borrow rate mismatch");
```

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Nitpicks then good to merge

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Nitpicks then good to merge

**Comment by lucas-manuel** [comment]
Also made a ticket here: https://linear.app/phoenixlabs/issue/SC-241/add-full-test-coverage-for-strategies-in-sparklend-advanced

Think this is a great start to improve test coverage, but want to add dedicated tests for calculateInterestRates as well.

Could also discuss removing stable logic altogether for these contracts.

File: test/InterestRateStrategyBase.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/InterestRateStrategyBase.t.sol]: done

File: test/InterestRateStrategyBase.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/InterestRateStrategyBase.t.sol:70]: done

**Comment by hexonaut** [comment]
Ya agree about removing the stable borrow logic. It's wasting gas at the very least. I can open a separate PR for this and your other suggestion soon.

**Review by lucas-manuel — APPROVED** [review]

**Review by barrutko — APPROVED** [review]

**Pull request merged**